### ISSUE TITLE
Strided slice implementation leads to negative tensor dimension 

### ISSUE THREAD

**Issue by 11mhg** [tensorflow/tfjs#4853]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow.js): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 20.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow.js installed from (npm or script link): yes
- TensorFlow.js version (use command below): 3.3.0
- Browser version: N/A
- Tensorflow.js Converter Version: 

```
tensorflowjs 3.3.0

Dependency versions:
  keras 2.4.0
  tensorflow 2.4.1
```

**Describe the current behavior**
After saving a model that contains tf.keras.applications.VGG19, I use tensorflowjs_converter to convert this model into a tfjs model. I load this model in as a graphModel and attempt to do some inferecing using `model.predict( x );` and I get the following error:

```
Error occured during script running/parsing:  Error: Tensor must have a shape comprised of positive integers but got shape [1,224,224,-3].
at assert (/content/node_modules/@tensorflow/tfjs-core/dist/tf-core.node.js:335:15)
at /content/node_modules/@tensorflow/tfjs-core/dist/tf-core.node.js:813:9
at Array.forEach (<anonymous>)
at assertNonNegativeIntegerDimensions (/content/node_modules/@tensorflow/tfjs-core/dist/tf-core.node.js:812:11)
at Object.buffer (/content/node_modules/@tensorflow/tfjs-core/dist/tf-core.node.js:5866:5)
at stridedSliceImpl (/content/node_modules/@tensorflow/tfjs-backend-cpu/dist/tf-backend-cpu.node.js:1654:27)
at Object.stridedSlice [as kernelFunc] (/content/node_modules/@tensorflow/tfjs-backend-cpu/dist/tf-backend-cpu.node.js:8314:22)
at kernelFunc (/content/node_modules/@tensorflow/tfjs-core/dist/tf-core.node.js:3139:32)
at /content/node_modules/@tensorflow/tfjs-core/dist/tf-core.node.js:3200:27
at Engine.scopedRun (/content/node_modules/@tensorflow/tfjs-core/dist/tf-core.node.js:3012:23)
```

**Describe the expected behavior**

The expected behaviour would be that the model predicts as normal and produces a tensor. 

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/CodePen/any notebook.

[https://colab.research.google.com/drive/1qPMyOuYyTlmyedh1I2pyDHC8WOheKKIE?usp=sharing](https://colab.research.google.com/drive/1qPMyOuYyTlmyedh1I2pyDHC8WOheKKIE?usp=sharing)

**Comment by rthadur** [comment]
looks like there is shape mismatch, you need provide correct shape.

**Comment by 11mhg** [comment]
Hi @rthadur ,

I believe you are wrong in this regard as the error message clearly states `Error occured during script running/parsing:  Error: Tensor must have a shape comprised of positive integers but got shape [1,224,224,-3].` and yet I have given it a tensor of shape `[1,224,224,3]`. You'll note that this is all positive integers and the correct shape. Please review the provided colab for more implementation details if need be.

**Comment by 11mhg** [comment]
This bug is still an issue and has not be addressed. Commenting to keep alive as it is definitely not stale as it has not been addressed.

**Comment by 11mhg** [comment]
I've managed to narrow this down to an issue with adding a lambda layer to the python model prior to js'ifying it.

that is to say
```
model = tf.keras.applications.VGG19(include_top=False, weights='imagenet', input_shape=(None,None,3))
input = tf.keras.layers.Input([224, 224, 3], batch_size=1, dtype = tf.float32)
preprocessor = tf.keras.applications.vgg19.preprocess_input
c = tf.keras.layers.Lambda(
    lambda x: preprocessor(x)
)(input)
output = model(c)
final_model = tf.keras.Model(inputs = input, outputs = out)
```
This model results in an issue with the strided slice and yet if one removes the lambda with that preprocessor, one gets a model that works fine when converted to tensorlfow js.


This is particularly strange because vgg19's preprocessing is simply reversing RGB to BGR and mean normalizing the pixels to imagenet average pixel colors...

**Comment by lina128** [comment]
Hi @11mhg, it seems we don't support lamda layer conversion, you will have to provide the js implementation of the lambda layer, see this issue for details: https://github.com/tensorflow/tfjs/issues/283
